### ISSUE TITLE
Dataset.image should be callable by element names

### ISSUE THREAD

**Issue by iliiliiliili** [LukasHedegaard/datasetops#24]
Dataset.image can be only called without parameters to convert all convertible data to image or with flags `(True, False, True)`. We should also be able to call it with names: `ds.image("image_2")`

**Comment by LukasHedegaard** [comment]
I agree with @iliiliiliili , that `ds.image("image_2")` is prettier than the flags.
The "positional" behaviour is currently used for multiple functions:
- `image`
- `numpy`
- `reshape`
- `image_resize`

We should decide between the "positional" behaviour (flags) or using keys.
To give an example, we're deciding between
```
ds = ds.named('label', 'image')
ds.reshape(False, new_shape)
ds.reshape(None, new_shape)
```

or 

```
ds = ds.named('label', 'image')
ds.reshape(1, new_shape)
ds.reshape([1], new_shape)
ds.reshape('image', new_shape)
ds.reshape(['image'], new_shape)
```

What do you think, @clegaard ?

**Comment by clegaard** [comment]
I strongly favor option 2 since it is more readable in my opinion.

**Comment by LukasHedegaard** [comment]
It seems we're in agreement. I'll include the fix alongside the new image transformations after Easter.
